src/v3/src/util/widgetHooks.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IdxStatus, IdxTransaction } from '@okta/okta-auth-js';
+
+import { HookFunction, HooksOptions, HookType } from '../../../types';
+import { getFormNameForTransaction } from './getEventContext';
+
+export class WidgetHooks {
+  private hooks: Map<string, Map<string, HookFunction[]>>;
+
+  /* eslint-disable no-restricted-syntax */
+  constructor(hooksOptions?: HooksOptions) {
+    this.hooks = new Map();
+    if (hooksOptions) {
+      for (const [formName, formHooks] of Object.entries(hooksOptions)) {
+        const formHooksMap = new Map<string, HookFunction[]>();
+        for (const [hookType, hooksByType] of Object.entries(formHooks)) {
+          if (hooksByType) {
+            formHooksMap.set(hookType, hooksByType);
```

Comment:
Can we use `this.addHook` instead here? Would be better to use the same public method to add hooks.

I also see the returned type of `hookType` and `hooksByType` is `any` right now so we should address that as well